### PR TITLE
Doc: add 'enable authentication and restriction' document

### PR DESCRIPTION
**Pull request by AlinsRan**
### Type of change:

- [ ] Bugfix
- [ ] New feature provided
- [ ] Improve performance
- [ ] Backport patches

### Pre-submission checklist:

* [ ] Did you explain what problem does this PR solve? Or what new features have been added?
* [ ] Have you added corresponding test cases?
* [ ] Have you modified the corresponding document?
* [ ] Is this PR backward compatible? **If it is not backward compatible, please discuss on the [mailing list](https://github.com/apache/apisix-ingress-controller#community) first**

### CODE REVIEW

File: docs/en/latest/practices/enable-authentication-and-restriction.md [hunk 1]
```diff
@@ -0,0 +1,342 @@
...
+    backends:
+    - serviceName: httpbin
+      servicePort: 80
+    authentication:
+      enable: true
+      type: keyAuth
+EOF
+```
+
+**Test keyAuth**
```
Comment (tao12345666333) [inline docs/en/latest/practices/enable-authentication-and-restriction.md:109]: I think an example without headers can be added here.

File: docs/en/latest/practices/enable-authentication-and-restriction.md [hunk 2]
```diff
@@ -0,0 +1,342 @@
...
+     servicePort: 80
+   authentication:
+     enable: true
+     type: keyAuth
+   plugins:
+   - name: consumer-restriction
+     enable: true
+     config:
+       whitelist:
+       - "default_jack1"
```
Comment (tao12345666333) [inline docs/en/latest/practices/enable-authentication-and-restriction.md:212]: Here you can add its generated rules

File: docs/en/latest/practices/enable-authentication-and-restriction.md [hunk 3]
```diff
@@ -0,0 +1,342 @@
...
+   backends:
+   - serviceName: httpbin
+     servicePort: 80
+   authentication:
+     enable: true
+     type: keyAuth
+   plugins:
+   - name: consumer-restriction
+     enable: false
+     config:
```
Comment (tao12345666333) [inline docs/en/latest/practices/enable-authentication-and-restriction.md:338]: The content in the original config field does not need to be deleted, just modify `enable: false`

**Review by tao12345666333 — COMMENTED** [review]
thanks!

**Comment by juzhiyuan** [comment]
Hi, @AlinsRan, please help to update your PR's checklist.

Also cc @navendu-pottekkat to have a review.

**Comment by AlinsRan** [comment]
Hi @navendu-pottekkat, Please help review.

File: docs/en/latest/practices/enable-authentication-and-restriction.md [hunk 4]
```diff
@@ -0,0 +1,362 @@
...
+
+## Description
+
+Consumers are useful when you have different consumers requesting the same API and you need to execute different Plugin and Upstream configurations based on the consumer. These need to be used in conjunction with the user authentication system.  
+
+## Attributes
+
+### Authentication
+
+| Name      | Type   | Requirement | Value                                                   |  Description                                                                                |
```
Comment (navendu-pottekkat) [inline docs/en/latest/practices/enable-authentication-and-restriction.md:32]: Could you change the table structure similar to this?

https://github.com/apache/apisix/pull/6877/files

File: docs/en/latest/practices/enable-authentication-and-restriction.md [hunk 5]
```diff
@@ -0,0 +1,362 @@
+---
+title: enable authentication and restriction
```
Comment (navendu-pottekkat) [inline docs/en/latest/practices/enable-authentication-and-restriction.md:2]: Use sentence case.

```suggestion
title: Enable authentication and restriction

```

File: docs/en/latest/practices/enable-authentication-and-restriction.md [hunk 6]
```diff
@@ -0,0 +1,362 @@
...
+```shell
+kubectl  exec -it -n ${namespace of Apache APISIX} ${pod of Apache APISIX}  -- curl http://127.0.0.1:9080/anything -H 'Host: local.httpbin.org' -H 'apikey:jack2-key' -d '' -v
+
+HTTP/1.1 403 Forbidden
+...
+```
+
+### Disable Plugin
+
+When you want to disable the `consumer-restriction` plugin, it is very simple,
```
Comment (navendu-pottekkat) [inline docs/en/latest/practices/enable-authentication-and-restriction.md:323]: Could you change this line to the disable plugin line in this PR? https://github.com/apache/apisix/pull/6877/files

**Review by navendu-pottekkat — COMMENTED** [review]
It would be consistent if you could structure this doc similar to the one shown here: https://github.com/apache/apisix/pull/6877/files

We are changing the docs to follow this structure. So it would be great if you could follow those headings and subheadings.

**Comment by tao12345666333** [comment]
@AlinsRan plz check the line error.

**Comment by AlinsRan** [comment]
> @AlinsRan plz check the lint error.
The lint of my repository is OK.
This error is `minikube.md`.

**Comment by AlinsRan** [comment]
> @AlinsRan plz check the lint error.

The lint of my repository is OK.  
This error is minikube.md.

**Review by tao12345666333 — COMMENTED** [review]
left a comment, the rest I think is OK

File: docs/en/latest/practices/enable-authentication-and-restriction.md [hunk 7]
```diff
@@ -0,0 +1,376 @@
...
+       allowed_by_methods:
+       - user: "default_jack1"
+         methods:
+         - "POST"
+         - "GET"
+       - user: "default_jack2"
+         methods:
+         - "GET"
+EOF
+```
```
[the remark this answers is not in the document]
Comment (AlinsRan) [reply docs/en/latest/practices/enable-authentication-and-restriction.md:376]: Sure.

**Review by tao12345666333 — APPROVED** [review]
LGTM

File: docs/en/latest/practices/enable-authentication-and-restriction.md [hunk 8]
```diff
@@ -0,0 +1,385 @@
...
+
+**Example usage**
+
+Requests from jack1:
+
+```shell
+kubectl  exec -it -n ${namespace of Apache APISIX} ${pod of Apache APISIX}  -- curl http://127.0.0.1:9080/anything -H 'Host: local.httpbin.org' -H 'apikey:jack1-key' -v
```
Comment (hf400159) [inline docs/en/latest/practices/enable-authentication-and-restriction.md:309]: Usually, the command and return result need to be separated. And for long commands, we need to format them. Please check the full text.

**Review by hf400159 — COMMENTED** [review]
LGTM

File: docs/en/latest/practices/enable-authentication-and-restriction.md (hunk 8 above)
Comment (AlinsRan) [reply docs/en/latest/practices/enable-authentication-and-restriction.md:309]: ok

**Comment by juzhiyuan** [comment]
And @navendu-pottekkat, also this one.

**Comment by tao12345666333** [comment]
ping @tokers  @navendu-pottekkat for review, thanks

**Review by tao12345666333 — APPROVED** [review]

File: docs/en/latest/practices/enable-authentication-and-restriction.md [hunk 9]
```diff
@@ -0,0 +1,404 @@
...
+    keyAuth:
+      value:
+        key: foo-key
+EOF
+```
+
+ApisixRoute:
+
+```shell
+kubectl apply -f -<<EOF
```
Comment (juzhiyuan) [inline docs/en/latest/practices/enable-authentication-and-restriction.md:127]: Hi, may I know why we have `EOF` here?

Also, cc @navendu-pottekkat @kwanhur to help review. 😉

Comment (AlinsRan) [reply docs/en/latest/practices/enable-authentication-and-restriction.md:127]: > 

It means that there after be anonymous files ending with EOF later.
Here, it is mainly used to shorten the length of the article.

Comment (AlinsRan) [reply docs/en/latest/practices/enable-authentication-and-restriction.md:127]: > Hi, may I know why we have `EOF` here?
> 
> Also, cc @navendu-pottekkat @kwanhur to help review. 😉
It means that there after be anonymous files ending with EOF later.
Here, it is mainly used to shorten the length of the article.

File: docs/en/latest/practices/enable-authentication-and-restriction.md [hunk 10]
```diff
@@ -0,0 +1,404 @@
...
+
+## Example
+
+### Prepare env
+
+Kubernetes cluster:
+
+1. [apisix-ingress-controller](../deployments/minikube.md).
+2. httpbin.
+
```
Comment (navendu-pottekkat) [inline docs/en/latest/practices/enable-authentication-and-restriction.md:97]: Could you make this clearer that we want readers to setup APISIX ingress and install HTTPBin on their cluster?

**Review by navendu-pottekkat — COMMENTED** [review]
You can mention the plugins being used in the first step and link every reference to a particular Plugin to its documentation page.

~Apisix~ to APISIX.

**Comment by tao12345666333** [comment]
ping @AlinsRan any update？

File: docs/en/latest/practices/enable-authentication-and-restriction.md [hunk 11]
```diff
@@ -0,0 +1,404 @@
...
+> ```shell
+> $ kubectl get apisixconsumers.apisix.apache.org -n default
+> NAME    AGE
+> foo     14h
+> jack1   14h
+> jack2   14h
+> ```
+>
+> `${consumer_name}` = `${namespace}_${ApisixConsumer_name}` --> `default_foo`  
+> `${consumer_name}` = `${namespace}_${ApisixConsumer_name}` --> `default_jack1`
```
[the remark this answers is not in the document]
Comment (AlinsRan) [reply docs/en/latest/practices/enable-authentication-and-restriction.md:246]: I tried. Doesn't support `Admonitions`.
How to change it?

**Comment by tao12345666333** [comment]
a kind reminder @navendu-pottekkat @tokers @juzhiyuan @hf400159

File: docs/en/latest/practices/enable-authentication-and-restriction.md (hunk 11 above)
[the remark this answers is not in the document]
Comment (navendu-pottekkat) [reply docs/en/latest/practices/enable-authentication-and-restriction.md:246]: The apisix-website repo supports admonitions. It should work. In this repo, preview is not available as it cannot be built.

File: docs/en/latest/practices/enable-authentication-and-restriction.md (hunk 10 above)
Comment (navendu-pottekkat) [reply docs/en/latest/practices/enable-authentication-and-restriction.md:97]: Yes. I agree that we shouldn't talk about how to install but I think we should make it clear that the **user have to install ingress controller and HTTPbin for the rest of the tutorial to work. Point to other documents on how to do that.

File: docs/en/latest/practices/enable-authentication-and-restriction.md (hunk 11 above)
[the remark this answers is not in the document]
Comment (AlinsRan) [reply docs/en/latest/practices/enable-authentication-and-restriction.md]: Solved!

File: docs/en/latest/practices/enable-authentication-and-restriction.md (hunk 10 above)
Comment (AlinsRan) [reply docs/en/latest/practices/enable-authentication-and-restriction.md]: Solved!

**Comment by juzhiyuan** [comment]
> a kind reminder @navendu-pottekkat @tokers @juzhiyuan @hf400159

@tao12345666333 Please help to check if the guide is correct except for Style. @navendu-pottekkat and I are discussing Doc Maintainers Program, and for this PR, if the contents are correct we could merge first.

**Comment by tao12345666333** [comment]
The guide is correct. 

I have approved it before

**Comment by tao12345666333** [comment]
One of our current problems is that we don't have enough approvers.

So I need your help.

**Review by juzhiyuan — APPROVED** [review]

**Comment by juzhiyuan** [comment]
Just checked, and LGTM.

**Pull request merged**